kornia/feature/disk/detector.py

```diff
@@ -0,0 +1,138 @@
+from typing import List, Optional, Tuple
+
+import torch
+import torch.nn.functional as F
+from torch.distributions import Bernoulli, Categorical
+
+from kornia.core import Tensor
+from kornia.utils import create_meshgrid
+
+from .structs import Keypoints
+
+
+def select_on_last(values: Tensor, indices: Tensor) -> Tensor:
+    '''
+    WARNING: this may be reinventing the wheel, but I don't know how to do
+    it otherwise with PyTorch.
+
+    This function uses an array of linear indices `indices` between [0, T] to
+    index into `values` which has equal shape as `indices` and then one extra
+    dimension of size T.
+    '''
+    return torch.gather(values, -1, indices[..., None]).squeeze(-1)
+
+
+def point_distribution(logits: Tensor) -> Tuple[Tensor, Tensor, Tensor]:
+    """Implements the categorical proposal -> Bernoulli acceptance sampling scheme.
+
+    Given a tensor of logits, performs samples on the last dimension, returning     a) the proposals     b) a binary
+    mask indicating which ones were accepted     c) the logp-probability of (proposal and acceptance decision)
+    """
+
+    proposal_dist = Categorical(logits=logits)
+    proposals = proposal_dist.sample()
+    proposal_logp = proposal_dist.log_prob(proposals)
+
+    accept_logits = select_on_last(logits, proposals).squeeze(-1)
+
+    accept_dist = Bernoulli(logits=accept_logits)
+    accept_samples = accept_dist.sample()
+    accept_logp = accept_dist.log_prob(accept_samples)
+    accept_mask = accept_samples == 1.0
+
+    logp = proposal_logp + accept_logp
+
+    return proposals, accept_mask, logp
+
+
+def nms(signal: Tensor, window_size=5, cutoff=0.0) -> Tensor:
```

Comment:
@jatentaki wouldn't `kornia.geometry.subpix.nms2d` work here?
https://github.com/kornia/kornia/blob/master/kornia/geometry/subpix/nms.py#L27

Reply:
True :) though my implementation seems more efficient - you're using conv2d with `eye` kernel and then max, while I use a single call to `max_pool2d`

Reply:
I think this worth at least open a issue to map these nms, and try to unify into the more efficient case  -- with this pr, i think will be at least the third nms inside kornia :)

Reply:
@johnnv1 they also work a bit differently. MaxPool-based (in DISK) returns 1 for torch.ones(3, 3). Conv-based (in subpix) allows only strict maximum